portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +32,66 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    private void addEntity(String text, long timestamp)
+    {
+        Entity commentEntity = new Entity("Comments");
+        commentEntity.setProperty("commentInput", text);
+        commentEntity.setProperty("timestamp", timestamp);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(commentEntity);
+    }
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Brandon Vu!</h1>");
+    private HashMap<String,ArrayList<String>> hashComment(PreparedQuery results, int nComments)
+    {
+        HashMap<String,ArrayList<String>> entityHash = new HashMap<>();
+        ArrayList<String> cArray = new ArrayList<>();
+        int count = 0;
+        for (Entity entity : results.asIterable()) 
+        {
+            if (nComments != count)
+            {
+            String eachComment = (String) entity.getProperty("commentInput");
+            cArray.add(eachComment);
+            entityHash.put("comments",cArray);
```

Comment:
Can this line `entityHash.put("comments",cArray);` be moved to be after the for-loop completes? It would be more efficient and clear that way, unless I misunderstood the logic.